lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -684,3 +685,23 @@ internal fun createUniqueExprValueFilter(): (ExprValue) -> Boolean {
 
     return { exprValue -> seen.add(exprValue) }
 }
+
+fun Sequence<ExprValue>.distinct(): Sequence<ExprValue> {
+    val seen = TreeSet(DEFAULT_COMPARATOR)
+    return sequence {
+        this@distinct.forEach {
+            if (!seen.contains(it)) {
+                seen.add(it.unnamedValue())
+                yield(it)
+            }
+        }
+    }
+}
+
+fun Sequence<ExprValue>.multiplicities(): TreeMap<ExprValue, Int> {
+    val multiplicities: TreeMap<ExprValue, Int> = TreeMap(DEFAULT_COMPARATOR)
+    this.forEach {
+        multiplicities.compute(it) { _, v -> (v ?: 0) + 1 }
+    }
+    return multiplicities
+}
```

Comment:
Are these intentionally public? If so, may need some kdoc describing these extension functions.

Reply:
`distinct()` yes, but `multiplicities()` could be private in `ExprValBagOp`. I can do the little refactor in a different PR